firmware/shared/src/hw/hw_sdio_sd.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "hw_hal.h"
+#include <stdbool.h>
+#include <stdint.h>
+
+// Check here for documentation about the type defined by HAL
+// https://www.disca.upv.es/aperles/arm_cortex_m3/llibre/st/STM32F439xx_User_Manual/structsd__handletypedef.html
+
+typedef struct
+{
+    SD_HandleTypeDef *const hsd;     // the HAL SD handle that will hold the state of the SD card
+    uint32_t                timeout; // the timeout for the SD card operations
+} SD_card;                           // struct that included all the state about SDIO and SD card
```

Comment:
We use `PascalCase` for type casing. So this should be `SdCard`, below is `SdCardConfig`, etc.